lib/pem.js

```diff
@@ -28,20 +28,40 @@
  * openssl req -x509 -nodes -days 365 -newkey rsa:2048 -sha1 -keyout private.pem -out public.pem
  */
 
+
+/*
+ * certificate string and private key string have the following format:
+ * 1. certificate string:
+ *    "---BEGIN CERTIFICATE---- content ---END CERTIFICATE---"
+ * 2. private key string:
+ *    "---BEGIN PRIVATE KEY---- content ---END PRIVATE KEY---"
+ *
+ * We define two regex (CERT_REGEX and PRIVATE_KEY_REGEX), in which the content 
+ * we are interested in is captured as a (and the only) capture group. We can 
+ * then use the getFirstCapturingGroup function below to get the content, and 
+ * use the removeRN function below to sanitize the content by removing the \r 
+ * and \n's in the content.
+ */
+
 /* eslint-disable max-len */
-const CERT_REGEX = /-+BEGIN\s+.*CERTIFICATE[^-]*-+(?:\s)+([a-zA-Z0-9+\/=\r\n]+)-+END\s+.*CERTIFICATE[^-]*-+/;
-const PRIVATE_KEY_REGEX = /-+BEGIN\s+.*PRIVATE\s+.*KEY[^-]*-+(?:\s)+([a-zA-Z0-9+\/=\r\n]+)-+END\s+.*PRIVATE\s+.*KEY[^-]*-+/;
+const CERT_REGEX = /-+BEGIN\s+.*CERTIFICATE[^-]*-+\s+([a-zA-Z0-9+\/=\r\n]+)-+END\s+.*CERTIFICATE[^-]*-+/;
```

Comment:
I recommend simplifying this regex to be stricter and easier to read. Suggest:
-+BEGIN\s+CERTIFICATE-+\s+([a-zA-Z0-9+\/=\r\n]+)\s+-+END\s+CERTIFICATE-+